src/js/common/schemaform/FormPage.jsx

```diff
@@ -54,12 +55,16 @@ class FormPage extends React.Component {
   }
 
   onChange(formData) {
-    this.props.setData(formData);
+    let newData = formData;
+    if (this.props.route.pageConfig.showPagePerItem) {
+      newData = _.set([this.props.route.pageConfig.arrayPath, this.props.params.index], formData, this.props.form.data);
```

Comment:
@cehsu Yeah, that's a good point, I should work that comment into comments into the actual code.

@cvalarida I'll add a comment, but the data that comes back here is just for a single row in the array, so we need to update the full form data object before saving it in the redux store.